src/pudl/analysis/allocate_net_gen.py

```diff
@@ -413,26 +406,46 @@ def _associate_fuel_type_only_wo_matching_fuel_type(gens_asst, gf_grouped):
              .sum(min_count=1)),
             on=idx_plant,
             how='left',
-            suffixes=('', '_unconnected')
+            suffixes=('', '_missing_pm')
         )
         .assign(
             net_generation_mwh_gf=lambda x:
                 x.net_generation_mwh_gf.fillna(
                     x.net_generation_mwh_fuel
-                    + x.net_generation_mwh_fuel_unconnected.fillna(0)
+                    + x.net_generation_mwh_fuel_missing_pm.fillna(0)
                 ),
             fuel_consumed_mmbtu=lambda x:
                 x.fuel_consumed_mmbtu.fillna(
                     x.fuel_consumed_mmbtu_fuel
-                    + x.fuel_consumed_mmbtu_fuel_unconnected.fillna(0)
+                    + x.fuel_consumed_mmbtu_fuel_missing_pm.fillna(0)
                 ),
         )
     )
     return gens_asst_w_unassociated
 
 
-def make_allocation_ratio(gens_asst):
-    """Generate a ratio to use to allocate net generation by."""
+def make_allocation_ratio(gens_asst, pudl_out):
+    """
+    Generate a ratio to use to allocate net generation by.
+
+    The goal of  this function is to generation a column called `gen_ratio`,
+    which will be a ratio to allocate net generation from the gf table from
+    each `IDX_PM_FUEL` groups.
+    """
+    gen_pm_fuel = prep_alloction_ratio(gens_asst)
+    gen_pm_fuel_ratio = calc_allocation_ratios(gen_pm_fuel, pudl_out)
+
+    return gen_pm_fuel_ratio
+
+
+def prep_alloction_ratio(gens_asst):
```

Comment:
What columns need to be in `gens_asst`? What state does it the data need to be in at this point?